### ISSUE TITLE
Wrong remote URL for webdav backend

### ISSUE THREAD

**Issue by FatemeKhodayari** [iterative/dvc.org#3044]
Hi. I'm a beginner to DVC. Yesterday, I set up DVC with a `webdav` backend as said in the official documentation. When I wanted to push to the remote, the URL had `00` added to it. It must have been `https://mydomain/webdav/models` but instead DVC was trying to push to `https://mydomain/webdav/models/00` which is a nonexisting URL.

Any idea why this is happening and how I should solve it?

**Comment by jorgeorpinel** [comment]
> Also asked in https://groups.google.com/a/iterative.ai/g/support/c/F4pPX7UVdvs/m/Yv1KGbWPCAAJ?utm_medium=email&utm_source=footer&pli=1

**Comment by FatemeKhodayari** [comment]
> > Also asked in https://groups.google.com/a/iterative.ai/g/support/c/F4pPX7UVdvs/m/Yv1KGbWPCAAJ?utm_medium=email&utm_source=footer&pli=1

It seems the content's not available. Anyway, after going through apache logs I realised the actual error is because of the auth scheme. DVC is using basic auth for webdav (and yes, you are not able to change it) while our webdav server had digest as auth scheme. So we had to change the auth scheme of the webdav server.

After that, the auth problem was solved but we were facing a new issue about redirection. The actual problem was with the trailing slash. DVC was sending a URL without the trailing slash, apache / webdav was redirecting it to the URL with the trailing slash but the URL was not being followed. So we were getting 301 as a response to our post (push) request. This one was fixed by `export FSSPEC_DAV_FOLLOW_REDIRECTS=true` which IMHO is not a good solution. It was way better if we could set it through something like `dvc config` or `dvc modify`.
